tests/gtest/avifdecodetest.cc

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <array>
+#include <tuple>
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace libavif {
+namespace {
+
+// Used to pass the data folder path to the GoogleTest suites.
+const char* data_path = nullptr;
+
+TEST(AvifDecodeTest, ColorGridAlphaNoGrid) {
```

Comment:
Nit/optional: Instead of disabling this test in AV2 only builds, this test can detect if no AV1 decoder is available and skip the test. We should be able to adapt the code in avifavmtest.cc (search for "SKIP") for this test.

```
TEST(AvmTest, Av1StillWorksWhenAvmIsEnabled) {
  const char* encoding_codec =
      avifCodecName(AVIF_CODEC_CHOICE_AUTO, AVIF_CODEC_FLAG_CAN_ENCODE);
  const char* decoding_codec =
      avifCodecName(AVIF_CODEC_CHOICE_AUTO, AVIF_CODEC_FLAG_CAN_DECODE);
  if (encoding_codec == nullptr || std::string(encoding_codec) == "avm" ||
      decoding_codec == nullptr || std::string(decoding_codec) == "avm") {
    GTEST_SKIP() << "AV1 codec unavailable, skip test.";
  }
```
@y-guyon Yannis: Looking at the code above, I think we may need to refine `AVIF_CODEC_FLAG_CAN_ENCODE` and  `AVIF_CODEC_FLAG_CAN_DECODE` to AV1 and AV2 specific flags. For example, we can define `AVIF_CODEC_FLAG_CAN_ENCODE` and `AVIF_CODEC_FLAG_CAN_DECODE` as AV1-only, and add new flags `AVIF_CODEC_FLAG_CAN_ENCODE_AV2` and `AVIF_CODEC_FLAG_CAN_DECODE_AV2` for AV2.

Reply:
> add new flags `AVIF_CODEC_FLAG_CAN_ENCODE_AV2` and `AVIF_CODEC_FLAG_CAN_DECODE_AV2` for AV2.

Sure, fine with me.

Reply:
Done, added a helper function for checking AV1 encoder/decoder availability.